### DISCUSSION TITLE
How to use set_drag_drop_payload from Python?

### DISCUSSION THREAD

**Discussion by johnor** [pthom/imgui_bundle#85] [Q&A] [answered]
First of all, thanks for a great library! I am currently porting an application and have a question on how to use set_drag_drop_payload with a payload from Python.

calling " imgui.set_drag_drop_payload("hello", "a", 1)" results in a TypeError: 

TypeError: set_drag_drop_payload(): incompatible function arguments. The following argument types are supported:
    1. (type: str, data: capsule, sz: int, cond: int = 0) -> bool

How can I create a capsule-object from a string or an integer?

**Comment by pthom** [comment, marked as answer]
> First of all, thanks for a great library!

Thank you!

> TypeError: set_drag_drop_payload(): incompatible function arguments. The following argument types are supported:
> 1. (type: str, data: capsule, sz: int, cond: int = 0) -> bool
> How can I create a capsule-object from a string or an integer?

I need to review the binding for this function, and this will require an adaptation to the code generator.

Its C++ signature is:

```cpp
bool ImGui::SetDragDropPayload(const char* type, const void* data, size_t data_size, ImGuiCond cond)
```

Basically, I will need to wrap `void * data` into a `py::object` so that it accepts any python object. I'm looking at this now.

**Reply by johnor** [reply]
Great! Thanks for the quick response.

**Reply by pthom** [reply]
Ok, I added this commit: https://github.com/pthom/imgui_bundle/commit/d53ae36cf72de2ac820056fc71c5f702ef94f409

It adds support for Drag and Drop in python. Unfortunately it cannot support any kind of object but only integer values, as the C++ api is too low level (i.e it copies the raw object memory, which cannot be done in python). 

If you want to transfer custom payload, you will have to pass  `id(your_payload)` and store a reference to the paylod somewhere else. 

See [python demo code here](https://github.com/pthom/imgui_bundle/commit/d53ae36cf72de2ac820056fc71c5f702ef94f409#diff-e35c1af53f9c65e67b70b03d18ab48331982845b4c31795fe5a571583badb6e4)

**Reply by pthom** [reply]
Note: I did not update the package on pypi. If you want to get this, you will need to [install from source](https://pthom.github.io/imgui_bundle/#_install_for_python)

**Reply by pthom** [reply]
> Note: I did not update the package on pypi. If you want to get this, you will need to [install from source](https://pthom.github.io/imgui_bundle/#_install_for_python).

I usually only update the pypi package once there is a sufficient set of modifications.
